### PR TITLE
feat: Update flags for AMM and disallow incoming

### LINKED ISSUE #520: Support featureDisallowIncoming amendment
There is a new amendment which adds 4 account flags and account set flags.

Adds four Account Flags:
1. lsfDisallowIncomingNFTOffer
2. lsfDisallowIncomingCheck
3. lsfDisallowIncomingPayChan
4. lsfDisallowIncomingTrustline

Adds four new AccountSet Flags:
1. asfDisallowIncomingNFTOffer
2. asfDisallowIncomingCheck
3. asfDisallowIncomingPayChan
4. asfDisallowIncomingTrustline

XRPLF/rippled#4336

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

The amendment `featureDisallowIncoming` adds 4 new account flags and the AMM account flag was updated to accommodate them.

### Context of Change

This fixes a bug where the AMM page was not showing for AMM accounts.

Closes #520

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Review by pdp2121 — APPROVED** [review]

File: src/containers/shared/transactionUtils.ts [hunk 1]
```diff
@@ -68,6 +68,10 @@ export const TX_FLAGS: Record<string, Record<number, string>> = {
 }
 
 export const ACCOUNT_FLAGS: Record<number, string> = {
+  15: 'asfDisallowIncomingTrustline',
+  14: 'asfDisallowIncomingPayChan',
+  13: 'asfDisallowIncomingCheck',
+  12: 'asfDisallowIncomingNFTOffer',
```
Comment (JST5000) [inline src/containers/shared/transactionUtils.ts:74]: Is there no 11?

File: src/rippled/lib/utils.js [hunk 2]
```diff
@@ -14,7 +14,11 @@ export const ACCOUNT_FLAGS = {
   0x00400000: 'lsfGlobalFreeze',
   0x00800000: 'lsfDefaultRipple',
   0x01000000: 'lsfDepositAuth',
-  0x02000000: 'lsfAMM',
+  0x04000000: 'lsfDisallowIncomingNFTOffer',
+  0x08000000: 'lsfDisallowIncomingCheck',
+  0x10000000: 'lsfDisallowIncomingNFTOffer',
+  0x20000000: 'lsfDisallowIncomingNFTOffer',
```
Comment (JST5000) [inline src/rippled/lib/utils.js:20]: Is this supposed to be `lsfDisallowIncomingNFTOffer` 3 times?

File: src/containers/shared/transactionUtils.ts (hunk 1 above)
Comment (ckniffen) [reply src/containers/shared/transactionUtils.ts:74]: 11 is commented out in the rippled source as being reserved for hooks later.  We can add it when we enhance hook support.  I quickly tried looking but could not find it official referenced in the docs for hooks.

File: src/rippled/lib/utils.js (hunk 2 above)
Comment (ckniffen) [reply src/rippled/lib/utils.js:20]: Woops.

File: src/containers/shared/transactionUtils.ts (hunk 1 above)
Comment (ckniffen) [reply src/containers/shared/transactionUtils.ts:74]:
```
/*  // reserved for Hooks amendment
    lsfTshCollect = 0x02000000,     // True, allow TSH collect-calls to acc hooks
*/
```

**Review by JST5000 — APPROVED** [review]
LGTM

**Review by khancode — APPROVED** [review]

**Pull request merged**